Basic/2DSpaceShooter/Assets/Editor/Tutorials/TutorialCallbacks.cs

```diff
@@ -0,0 +1,61 @@
+﻿using UnityEngine;
+using UnityEditor;
+using Unity.Tutorials.Core.Editor;
+
+/// <summary>
+/// Implement your Tutorial callbacks here.
+/// </summary>
+[CreateAssetMenu(fileName = DefaultFileName, menuName = "Tutorials/" + DefaultFileName + " Instance")]
+public class TutorialCallbacks : ScriptableObject
+{
+    /// <summary>
+    /// The default file name used to create asset of this class type.
+    /// </summary>
+    public const string DefaultFileName = "TutorialCallbacks";
+
+    /// <summary>
+    /// Creates a TutorialCallbacks asset and shows it in the Project window.
+    /// </summary>
+    /// <param name="assetPath">
+    /// A relative path to the project's root. If not provided, the Project window's currently active folder path is used.
+    /// </param>
+    /// <returns>The created asset</returns>
+    public static ScriptableObject CreateAndShowAsset(string assetPath = null)
+    {
+        assetPath = assetPath ?? $"{TutorialEditorUtils.GetActiveFolderPath()}/{DefaultFileName}.asset";
+        var asset = CreateInstance<TutorialCallbacks>();
+        AssetDatabase.CreateAsset(asset, AssetDatabase.GenerateUniqueAssetPath(assetPath));
+        EditorUtility.FocusProjectWindow(); // needed in order to make the selection of newly created asset to really work
+        Selection.activeObject = asset;
+        return asset;
+    }
+
+    /// <summary>
+    /// Example callback for basic UnityEvent
+    /// </summary>
+    public void ExampleMethod()
+    {
+        Debug.Log("ExampleMethod");
+    }
+
+    /// <summary>
+    /// Example callbacks for ArbitraryCriterion's BoolCallback
+    /// </summary>
+    /// <returns></returns>
+    public bool DoesFooExist()
+    {
+        return GameObject.Find("Foo") != null;
+    }
+
+    /// <summary>
+    /// Implement the logic to automatically complete the criterion here, if wanted/needed.
+    /// </summary>
+    /// <returns>True if the auto-completion logic succeeded.</returns>
+    public bool AutoComplete()
+    {
+        var foo = GameObject.Find("Foo");
+        if (!foo)
+            foo = new GameObject("Foo");
+        return foo != null;
+    }
```

Comment:
This looks like example code. Is this used somewhere? If not, we can remove it.

Reply:
Apparently the TutorialCallbacks.asset throws a null reference exception when the scripts gets removed so I think it is needed?

```
NullReferenceException: Object reference not set to an instance of an object
UnityEditor.UIElements.Bindings.SerializedObjectBindingToBaseField`2[TValue,TField].FieldValueChanged (UnityEngine.UIElements.ChangeEvent`1[T] evt) (at <db4fb00fe00d4a8a850f8a745177ee1c>:0)
UnityEngine.UIElements.EventCallbackFunctor`1[TEventType].Invoke (UnityEngine.UIElements.EventBase evt) (at <dc61ba37583d47bc957110bffa4cd12c>:0)
UnityEngine.UIElements.EventCallbackRegistry+DynamicCallbackList.Invoke (UnityEngine.UIElements.EventBase evt, UnityEngine.UIElements.BaseVisualElementPanel panel, UnityEngine.UIElements.VisualElement target) (at <dc61ba37583d47bc957110bffa4cd12c>:0)
UnityEngine.UIElements.EventDispatchUtilities.HandleEventAcrossPropagationPath (UnityEngine.UIElements.EventBase evt, UnityEngine.UIElements.BaseVisualElementPanel panel, UnityEngine.UIElements.VisualElement target, System.Boolean isCapturingTarget) (at <dc61ba37583d47bc957110bffa4cd12c>:0)
UnityEngine.UIElements.EventDispatchUtilities.DefaultDispatch (UnityEngine.UIElements.EventBase evt, UnityEngine.UIElements.BaseVisualElementPanel panel) (at <dc61ba37583d47bc957110bffa4cd12c>:0)
UnityEngine.UIElements.EventBase.Dispatch (UnityEngine.UIElements.BaseVisualElementPanel panel) (at <dc61ba37583d47bc957110bffa4cd12c>:0)
UnityEngine.UIElements.EventDispatcher.ProcessEvent (UnityEngine.UIElements.EventBase evt, UnityEngine.UIElements.BaseVisualElementPanel panel) (at <dc61ba37583d47bc957110bffa4cd12c>:0)
UnityEngine.UIElements.EventDispatcher.ProcessEventQueue () (at <dc61ba37583d47bc957110bffa4cd12c>:0)
UnityEngine.UIElements.EventDispatcher.OpenGate () (at <dc61ba37583d47bc957110bffa4cd12c>:0)
```

Reply:
I was referring to the 3 methods specifically, the script is needed

Reply:
Ah I see ok fair enough, I am gonna shorten the script